src/main/java/com/onaio/steps/activities/SettingsActivity.java

```diff
@@ -191,4 +203,25 @@ public boolean isJustOpened() {
     public void setJustOpened(boolean justOpened) {
         this.justOpened = justOpened;
     }
+
+    public void prepareAvailableFormList() {
+
+        try {
+            List<ODKBlankForm> forms = ODKBlankForm.get(this, null);
+
+            int[] actvIdList = new int[] {R.id.form_id_household, R.id.form_id_participant};
+            for (int j : actvIdList) {
+                AutoCompleteTextView actvFormId = findViewById(j);
+                actvFormId.setThreshold(1);
+                actvFormId.setAdapter(new ArrayAdapter<>(this, android.R.layout.simple_dropdown_item_1line, forms));
+                actvFormId.setOnItemClickListener((adapterView, view, i, l) -> {
+                    ODKBlankForm selection = (ODKBlankForm) adapterView.getItemAtPosition(i);
+                    actvFormId.setText(selection.getJrFormId());
+                    actvFormId.setSelection(actvFormId.length());
+                });
+            }
+        } catch (AppNotInstalledException e) {
```

Comment:
I think the user should get a prompt for downloading ODK Collect at this point.

Reply:
this method is called every time from `onResume()` method in `SettingActivity` and alert is not suitable at that moment. Therefore, we have proper handling of this exception is available on the member detail page. Form checking here is supposed to be optional.